### PR TITLE
[SPARK-11970] [SQL] Add function SHOW to Dataset APIs

### PR DESCRIPTION
**Pull request by gatorsmile**
Add the function show() to Dataset APIs. The other functions were submitted in separate PRs. 

@marmbrus @rxin @cloud-fan Could you help me review the code changes?

Thank you!

### CODE REVIEW

**Comment by rxin** [comment]
Can you put this together with your join and sample changes?

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 1]
```diff
@@ -599,6 +599,38 @@ class Dataset[T] private[sql](
    */
   def takeAsList(num: Int): java.util.List[T] = java.util.Arrays.asList(take(num) : _*)
 
+  /**
+   * Compose the string representing rows for output
+   * @param _numRows Number of rows to show
+   * @param truncate Whether truncate long strings and align cells right
+   */
+  override private[sql] def showString(_numRows: Int, truncate: Boolean = true): String = {
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: can't we share the code between dataset and dataframe?

Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: When merging them into the same one, I hit a problem when calling the function `take`. The `Dataset` API has different type `Array[T]`, but `Dataframe` API has a type `Array[T]`. 

Do you have a better idea how to integrate into the same function?

**Comment by gatorsmile** [comment]
Sure, I can combine them into the same one when you think the implementation of `show` is ok. 

Thank you!

**Comment by rxin** [comment]
I implemented show as part of this patch: https://github.com/apache/spark/pull/9964

I just used toDF().show(). Can you close this pull request?

**Comment by gatorsmile** [comment]
Yeah, that is also my solution in my original solution in the PR for EXPLAIN. 
As long as you do not think they are duplicate codes, I like it. : )

**Pull request closed without merging**